tests/gdk/commands/component/test_PublishCommand.py

```diff
@@ -0,0 +1,827 @@
+            mock_file.assert_called_once_with(file_name, "w")
+            mock_json_dump.call_count == 1
+            assert mock_yaml_dump.called
+        assert "publish_recipe_file" in publish.project_config
+
+    def test_update_and_create_recipe_file_no_manifests(self):
+        self.mocker.patch("gdk.commands.component.project_utils.parse_recipe_file", return_value={})
+        mock_create_publish_recipe = self.mocker.patch.object(PublishCommand, "create_publish_recipe_file", return_value=None)
+
+        component_name = "component_name"
+        component_version = "1.0.0"
+        publish = PublishCommand({})
+        publish.update_and_create_recipe_file(component_name, component_version)
+        assert not mock_create_publish_recipe.called  # No 'Manifests' in recipe
+
+    def test_update_and_create_recipe_file_manifests_build(self):
+        self.mocker.patch(
+            "gdk.commands.component.project_utils.parse_recipe_file",
+            return_value=self.mock_get_proj_config.return_value["parsed_component_recipe"],
+        )
+        mock_glob = self.mocker.patch("pathlib.Path.glob", return_value=[Path("hello_world.py")])
+        mock_create_publish_recipe = self.mocker.patch.object(PublishCommand, "create_publish_recipe_file", return_value=None)
+
+        component_name = "com.example.HelloWorld"
+        component_version = "1.0.0"
+        publish = PublishCommand({})
+        publish.update_and_create_recipe_file(component_name, component_version)
+        publish_component_recipe = self.mock_get_proj_config.return_value["parsed_component_recipe"]
+        assert mock_glob.call_count == 1
+        assert mock_create_publish_recipe.call_count == 1
+        mock_create_publish_recipe.assert_any_call(component_name, component_version, publish_component_recipe)
+        assert publish_component_recipe["ComponentVersion"] == component_version
+        assert (
+            publish_component_recipe["Manifests"][0]["Artifacts"][0]["URI"]
+            == f"s3://default/{component_name}/{component_version}/hello_world.py"
+        )
+
+    def test_update_and_create_recipe_file_manifests_not_build(self):
+        self.mocker.patch(
+            "gdk.commands.component.project_utils.parse_recipe_file",
+            return_value=self.mock_get_proj_config.return_value["parsed_component_recipe"],
+        )
+        mock_create_publish_recipe = self.mocker.patch.object(PublishCommand, "create_publish_recipe_file", return_value=None)
+
+        component_name = "component_name"
+        component_version = "1.0.0"
+        publish = PublishCommand({})
+        with pytest.raises(Exception) as e:
+            publish.update_and_create_recipe_file(component_name, component_version)
+        assert "as it is not build.\nBuild the component `gdk component build` before publishing it." in e.value.args[0]
+        assert mock_create_publish_recipe.call_count == 0
+
+    def test_update_and_create_recipe_file_uri_not_matches(self):
```

Comment:
maybe add a test for `NEXT_PATCH` when there are no prior versions

Reply:
`test_get_next_version_component_not_exists` checks if 1.0.0 is used as the fallback version when a component with no previous versions is published with `NEXT_PATCH`